File: algorithms/1945.sum-of-digits-of-string-after-convert/solution.rs
impl Solution {
    pub fn get_lucky(s: String, k: i32) -> i32 {
        let mut sum = 0;

        for c in s.chars() {
            let mut n = (c as usize - 'a' as usize + 1) as i32;
            while n != 0 {
                sum += n % 10;
                n /= 10;
            }
        }

        for _ in 1..k {
            let mut curr = sum;
            sum = 0;

            while curr != 0 {
                sum += curr % 10;
                curr /= 10;
            }
        }

        sum
    }
}

File: algorithms/1945.sum-of-digits-of-string-after-convert/solution.go
func getLucky(s string, k int) int {
	var sum int

	for _, r := range s {
		n := int(r - 'a' + 1)
		for n != 0 {
			sum += n % 10
			n /= 10
		}
	}

	for i := 1; i < k; i++ {
		curr := sum
		sum = 0

		for curr != 0 {
			sum += int(curr % 10)
			curr /= 10
		}
	}

	return sum
}
